crates/walrus-stress/src/generator/blob.rs

```diff
@@ -31,30 +62,34 @@ impl BlobData {
         Self {
             bytes,
             rng,
-            min_size_log2,
-            max_size_log2,
+            epoch_to_store: config.get_random_epoch_to_store(),
+            config,
         }
     }
 
+    /// Returns the number of epochs to store the blob for.
+    pub fn epoch_to_store(&self) -> EpochCount {
+        self.epoch_to_store
+    }
+
     /// Changes the blob by incrementing (wrapping) a randomly chosen byte excluding
     /// the `TESTBLOB` tag at the start of the blob data.
+    ///
+    /// Also updates the epoch to store to a new random value.
     pub fn refresh(&mut self) {
         let index = self.rng.gen_range(TAG.len()..self.bytes.len());
         self.bytes[index] = self.bytes[index].wrapping_add(1);
+        self.epoch_to_store = self.config.get_random_epoch_to_store();
     }
 
     /// Returns a slice of the blob with a size `2^x`, where `x` is chosen uniformly at random
     /// between `min_size_log2` and `max_size_log2`.
     pub fn random_size_slice(&self) -> &[u8] {
-        let blob_size_log2 = thread_rng().gen_range(self.min_size_log2..=self.max_size_log2);
+        let blob_size_log2 =
+            thread_rng().gen_range(self.config.min_size_log2..=self.config.max_size_log2);
         let blob_size = 2_usize.pow(blob_size_log2 as u32);
```

Comment:
Unrelated to this change:
```suggestion
        let mut blob_size = 2_usize.pow(blob_size_log2 as u32);
        blob_size = (blob_size - 1) & thread_rng().gen_range(1..blob_size);
```
By doing this we could get more varied sizes.